client/src/featureform/resources.py

```diff
@@ -1389,6 +1389,10 @@ def get(self, stub) -> "FeatureVariant":
     def _create(self, stub) -> Optional[str]:
         if hasattr(self.source, "name_variant"):
             self.source = self.source.name_variant()
+
+        # TODO: remove this hack
+        if self.provider == "local-mode":
```

Comment:
can we not make it optional for local-mode?

Reply:
we have a really annoying line of code in register that says 

```
        if len(features) > 0 and inference_store == "":
            inference_store = "local-mode"
```

i got rid of that and it broke all local dash tests

Reply:
so would `inference_store == ""` for hosted if they don't want to use inference_store?

Reply:
it would, but how do you tell we're local or not? do we have a is_local mode flag?